src/handler.py

```diff
@@ -64,6 +62,14 @@ async def text_handler(update: Update, context: ContextTypes.DEFAULT_TYPE, con:
         (chat_id, user_id, text),
     )
     con.commit()
+
+
+async def generate_response(update: Update, con: psycopg2.connect) -> None:
```

Comment:
This function only needs the `chat_id`. The response should be delivered either by returning the generated text, or via a lambda, as I described earlier.